### ISSUE TITLE
build for r6530 on Solaris doesn't reference -lm properly (migrated from Trac #1033)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2586]
Original ticket http://projects.scipy.org/numpy/ticket/1033
Reported 2009-03-02 by trac user gbburkhardt, assigned to unknown.

The umath.so library references fmodf, et al, resulting in:

ImportError: ld.so.1: python: fatal: relocation error: file /opt/gnu/python-2.5.2/lib/python2.5/site-packages/numpy/core/umath.so: symbol fmodf: referenced symbol not found

Record of build and run of test is attached.

**Comment by thouis** [comment]
Attachment in Trac by trac user gbburkhardt, 2009-03-02: [r6530-build-test](http://projects.scipy.org/numpy/attachment/ticket/1033/r6530-build-test)

**Comment by thouis** [comment]
Comment in Trac by atmention:cournape, 2009-03-02

Should be fixed in r6533

**Comment by thouis** [comment]
Comment in Trac by trac user gbburkhardt, 2009-03-02

rebuild with r6535, installed, but:

glenn.burkhardt $ python -c 'import numpy; numpy.test(verbose=2)'
Running unit tests for numpy
NumPy version 1.3.0.dev
NumPy is installed in /opt/gnu/python-2.5.2/lib/python2.5/site-packages/numpy
Python version 2.5.2 (r252:60911, Nov 18 2008, 17:01:44) [GCC 4.3.1]
nose version 0.10.4

---

Ran 0 tests in 0.043s

OK

**Comment by thouis** [comment]
Comment in Trac by atmention:cournape, 2009-03-03

Did you install with east_install ? One common reason for nose not to find any test is that the corresponding test*py files are executable, which nose refuse to run for tests.

**Comment by thouis** [comment]
Comment in Trac by trac user gbburkhardt, 2009-03-03

For build and install, I am using:

python setup.py build --fcompiler=gnu95
python setup.py install

I changed the permissions, and the test ran:

numpy $ pwd
/opt/gnu/python-2.5.2/lib/python2.5/site-packages/numpy
numpy $ find . -name "test*.py" -exec chmod -x {} \;

But there's a new seg fault:

... lines omitted...
Tests median w/ 3D ... ok
Tests notmasked_contiguous ... ok
Tests unmasked_edges ... ok
Tests polyfit ... ok
test_atleast1d (test_extras.TestShapeBase) ... ok
Test creation by view ... ok
Test that 'exotic' formats are processed properly ... ok
Test filling the array ... ok
Tests fields retrieval ... Segmentation Fault (core dumped)

glenn.burkhardt $ gdb /opt/gnu/python-2.5.2/bin/python core
GNU gdb 6.8
Copyright (C) 2008 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later http://gnu.org/licenses/gpl.html
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "sparc-sun-solaris2.9"...
Core was generated by `python -c import numpy; numpy.test(verbose=2)'.
Program terminated with signal 11, Segmentation fault.
[New process 82357    ]
#0  0xfef48ec8 in _copy_from_same_shape (dest=0x1558690, src=0x99e718, 
    myfunc=0xfef22e10 <_unaligned_strided_byte_copy>, swap=0)
    at numpy/core/src/arrayobject.c:972
972         if ((dit == NULL) || (sit == NULL)) {
(gdb) p dit
No symbol "dit" in current context.
(gdb) p sit
$1 = (PyArrayIterObject *) 0x15df5a0
(gdb) l
967         sit = (PyArrayIterObject *)
968             PyArray_IterAllButAxis((PyObject *)src, &maxaxis);
969
970         maxdim = dest->dimensions[maxaxis];
971
972         if ((dit == NULL) || (sit == NULL)) {
973             Py_XDECREF(dit);
974             Py_XDECREF(sit);
975             return -1;
976         }
(gdb) p maxaxis
$2 = -1

I recompiled without optimization, and ran under 'gdb'.  The log 'numpy-segfault-3.3.09' is attached.

**Comment by thouis** [comment]
Attachment in Trac by trac user gbburkhardt, 2009-03-03: [numpy-segfault-3.3.09](http://projects.scipy.org/numpy/attachment/ticket/1033/numpy-segfault-3.3.09)

**Comment by thouis** [comment]
Comment in Trac by trac user gbburkhardt, 2009-03-03

The original problem for this ticket was fixed.  I'll put my new notes about the seg fault back in ticket 1000, since it's really about the unit tests not working under Solaris.

Perhaps the standard install on a Solaris system should fix the permissions for the files 'test*.py' when they're copied over.  I notice that the files are marked executable after extracting them from the zip file Trac created for me when I pulled a snapshot.
